mujoco_ros2_control/src/mujoco_ros2_control.cpp

```diff
@@ -42,19 +40,54 @@ MujocoRos2Control::~MujocoRos2Control()
   stop_cm_thread_ = true;
   cm_executor_->remove_node(controller_manager_);
   cm_executor_->cancel();
-  cm_thread_.join();
 }
 
-void MujocoRos2Control::init()
+std::string MujocoRos2Control::get_robot_description()
 {
-  clock_publisher_ = node_->create_publisher<rosgraph_msgs::msg::Clock>("/clock", 10);
+  // Getting robot description from parameter first. If not set trying from topic
+  std::string robot_description;
+
+  auto node = std::make_shared<rclcpp::Node>(
+    "robot_description_node",
+    rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true));
+  cm_executor_->add_node(node);
+
+  if (node->has_parameter("robot_description"))
+  {
+    robot_description = node->get_parameter("robot_description").as_string();
+    return robot_description;
+  }
+
+  RCLCPP_WARN(
+    logger_, "Impossible to get robot_description from parameter. Getting it from topic...");
```

Comment:
Totally optional but just for added clarity

```suggestion
  RCLCPP_WARN(
    logger_,
    "Failed to get robot_description from parameter. Will listen on the ~/robot_description "
    "topic...");
```